lightning/src/ln/onion_utils.rs

```diff
@@ -381,224 +396,242 @@ pub(super) fn build_first_hop_failure_packet(shared_secret: &[u8], failure_type:
 /// Returns update, a boolean indicating that the payment itself failed, the short channel id of
 /// the responsible channel, and the error code.
```

Comment:
nit: pre-existing, this comment is missing that it also returns the error message bytes, also it swaps the ordering of the bool and the short channel id